infrastructure/channels/serialize-data.js

```diff
@@ -0,0 +1,38 @@
+
+let cyclicArray = [2];
+cyclicArray.push(cyclicArray);
+
+let cyclicObject = {key1: "data"};
+cyclicObject.key2 = cyclicObject;
+
+let cyclicSet = new Set([1]);
+cyclicSet.add(cyclicSet);
+
+let cyclicMap = new Map([["key1", 1]]);
+cyclicMap.set("key2", cyclicMap);
+
+const objects = {
+    "null": {input: null},
+    "undefined": {input: undefined},
+    "int": {input: 1},
+    "Infinity": {input: Infinity},
+    "-Infinity": {input: -Infinity},
+    "NaN": {input: NaN},
+    "string": {input: "foo"},
+    "true": {input: true},
+    "false": {input: false},
+    "bigint": {input: 1n},
+    "RegExp": {input: /abc/g},
+    "Date": {input: new Date('December 17, 1995 03:24:00')},
```

Comment:
Someone's birthday? :)

Reply:
Presumably lots of people's, but I don't (think I) know any of them (I think I just copied it from somewhere).